### ISSUE TITLE
Shall we use the Software Carpentry workshop template for LC workshops?

### ISSUE THREAD

**Issue by weaverbel** [LibraryCarpentry/librarycarpentry.github.io#24]
I use this template for all the workshops I teach whether they are SWC or LC : https://github.com/swcarpentry/workshop-template 

To see a live one, look at:
https://weaverbel.github.io/2017-05-11-lc-boston/

That was a big customisation job as I had to change the software list, tweak the schedule etc. The good news is people could just copy and paste the software setup bit of my code into their own workshop template. SWC has good instructions on how to use the template - it's an import, not a fork, and you then get a gh-pages link that can be imported holus-bolus into AMY and it brings all the info in - dates, trainers, helpers, locations, lat/long, contact, etc. 

I will raise an issue on the SWC workshop template repo to ask that the LC logo also be an option - at the moment it is DC or SWC only. but workshops in AMY can be categorised as LC. 

I am an AMY admin so can help people get stuff on there. It provides a searchable archived list of workshops - so we could find all the LC workshops taught in the UK or all the ones taught by me or all the ones in a date range and so on. better than Google Docs in that regard - more permanent,

I am happy to walk people through a workshop template create/customisation via a zoom or hangout as it can be tricky the first time. I am an exceptionally old hand at this now - Git has no more terrors for me ;-)

**Comment by jezcope** [comment]
Given the amount of customisation needed, is it worth having our own fork (an actual fork rather than an import) of it as a starting point for LC workshops, but making sure we merge changes from the upstream SWC template regularly?

**Comment by weaverbel** [comment]
@jezcope Not a bad idea - but I don't 100% know how to do the merge - do you? We could have the fork in data-lessons and call it lc-workshop-template.

**Comment by jt14den** [comment]
Guys, I did a bit of the customization needed and mentioned by @weaverbel (added the logo option & library carpentry links in the _config.yml  for the upcoming LC at Pitt: 

https://datascienceinlibraries.github.io/2017-05-18-pitt/

I'd be happy to retrace my steps a bit to capture just the changes for a fork as what @jezcope describes. Maybe we can add that to the what Belinda has already one.

**Comment by jt14den** [comment]
I forked the SWC workshop template and added in the customizations from @weaverbel mixed with some of the ones I had and put up here: 

https://github.com/jt14den/workshop-template

The rendered website is here: http://www.tim-dennis.com/workshop-template/

Edit suggestions/changes welcome. 

Would it be better to live in the data-workshops or librarycarpentry organization? I'm not a collaborator on those orgs, so will need some help.

**Comment by drjwbaker** [comment]
@jt14den Added you to librarycarpentry.github.io I'm not 'Core' on https://github.com/orgs/data-lessons/teams so can't do that. Repo management is something we need to think about..

**Comment by drjwbaker** [comment]
I should add: template looks perfect. Is the workflow converting the details entered into a database of lessons (past, future) part of the template?

**Comment by jt14den** [comment]
@drjwbaker Ah, I typically email the admin@software-carpentry and they get it listed, but that's not spelled out. I also noticed that the instructions above the section 'general information' starting with 'This is a workshop template' contain a broken link to a dead workshop request form.  The SWC [request form](https://amy.software-carpentry.org/forms/swc/request/) does have a 'self-organized' radio button, but it's a Software Carpentry specific request form, so linking to that might confuse people. @weaverbel do you think we can add instructions to email admin@software-carpentry in our forked template and they post the workshop to Amy or would you rather be involved?  BTW, how do you go about getting Amy admin status? Is that for board members? I'd gladly help with this workflow if it is something I can do.

**Comment by weaverbel** [comment]
@jt14den @drjwbaker I think I should check with SWC about how to proceed. I am on the Software Carpentry steering committee so that is how I got access to AMY - I wrangle workshop requests for half of Australia to save SWC having to do the admin for all those. But AMY access is not given out to many people otherwise. I am happy to get into AMY any workshops we run, but they can only be AMY'ed if there is a workshop website in a particular address format like https://weaverbel.github.io/2017-05-11-lc-boston/ 

I think we should decide as LC if we want to use a workshop template (I would say yes, as you can do so much through it - software install instructions, workshop locations, who is teaching, getting on SWC map, tracking later through AMY, using eventbrite to book tickets, etc) but others may think differently. I do think it's a timesaver though as it just has everything on it you need to think about/organise.

Then, if we agree to go ahead, we check with SWC as to how they want to do the LC workshop template (LC fork?) - I can raise that at SWC SC - there is a meeting on 3 April if we can decide in time. 

Thoughts?

Copying to @jezcope and @ostephens too as they may have views.

**Comment by weaverbel** [comment]
@jt14den @drjwbaker @ostephens @jezcope 
+1 to James's point about repo ownership - we need to get that sorted, and maybe get a written procedure for adding new people as I feel I have to figure it out anew each time :-( 

The other issue is the survey links from the template. These are for SWC surveys and each new workshop has to be added to SurveyMonkey so that respondents find their specific workshop listed when they go to answer it. I removed them for the Boston one as they ask questions that are not 100% relevant and might create false expectations of what we will cover. I think it would be better to create our own surveys along the lines of what we did for Boston, and then people can add their own survey links to the template as they edit it for their own workshop.

**Comment by drjwbaker** [comment]
Here is where I am. I am easy on whether or not we use the SWC form as I have no experience of its functionality or usability. What we need is a way for people to advertise LC workshops at a URL and a way for listing past and future LC workshops at a URL. If we can easily combine these two needs via the SWC process, great. If it proves a faff, let's seek alternatives.

**Comment by weaverbel** [comment]
We have liftoff from SWC SC on customising the template to work for Library Carpentry. I am contacting someone to get that done and it should be available soon.

**Comment by jt14den** [comment]
@weaverbel yay! Great to hear.

**Comment by weaverbel** [comment]
Thanks @jt14den I have asked @rgaiacs to action this, and I've given him links to your and my template tweaks so I hope it will be soon. Thanks Raniere in advance!

**Comment by rgaiacs** [comment]
Initial work at https://github.com/swcarpentry/workshop-template/pull/393/.

**Comment by jezcope** [comment]
@jt14den Looks great! A couple of things we probably need to consider:

- `nano` is the default text editor for SWC, and the best justification for this I've heard is that it will still be available when you log into your HPC over SSH. Given that many fewer library folk will need to do this, should we consider recommending Atom/Sublime/Notebook++ instead? My feeling is that for this audience the more intuitive tool will be better.
- The current survey links go to the SWC pre-/post-workshop surveys: do we want to customise these for LC purposes too?

**Comment by ostephens** [comment]
@jezcope Emacs! (just kidding)

**Comment by ostephens** [comment]
@jezcope out of those I think that Atom is the only one that is free and available for Windows, Mac and Linux?

**Comment by jezcope** [comment]
@ostephens Atom was my thought too (Emacs might be a little bit too scary... :stuck_out_tongue_winking_eye:). Queue endless discussions about whether or not we should be advocating a specific software stack...

**Comment by drjwbaker** [comment]
I've had workshop attendees before complain that Atom is massive to download/install. For the record I use TextWrangler for Mac, gedit for Linux, Notepad++ for Windows.

**Comment by rgaiacs** [comment]
As far as I know, Atom is one of the text editor that satisfies the open source criteria and be available for Windows, Mac and Linux. Another one is Emacs, which has a [portable version for Windows](http://esnm.sourceforge.net/EmacsPortable.html). If you need to use a text editor I would recommend the same one for all the operating system since some learners are more visual.

**Comment by weaverbel** [comment]
@jezcope @drjwbaker @ostephens I am anti-Atom (though it is awesome) as (1) it is a huge download for something called Atom (!) and (2) while it has great functionality, it is probably too much cognitive overload for people to get up up and running with it quickly - there has been a lot of discussion on this issue on the SWC discuss list. I think notepad, gedit etc are better alternatives because they are simpler at first view. People find the shell a bit scary - we don't want to add more scary stuff!

**Comment by drjwbaker** [comment]
I too am anti-scaring people with tech just to fulfill open source criteria. If we can, I'd prefer to help people take baby steps and win them over gradually to open source goodness rather than go for shock and awe and lose them entirely.

**Comment by ostephens** [comment]
What is it about Atom that leads to cognitive overload? It seems straightforward to me to just edit a text file? Surely compared to nano it is much more user friendly?

Gedit is smaller, but just tried to install and the latest version didn't work on macOS for me (plus the package is unsigned, so there are extra hoops to jump through to install on Mac)

Not familiar with the latest version of Notepad, but is that still Windows only? Also in the past have had issues with line endings and encoding. So unless that's changed, I'm not in favour.

I use SublimeText which is cross-platform but not free (although you are allowed to install for free for evaluation purposes). Not Open Source. I don't see this as a simpler option than Atom though.

**Comment by weaverbel** [comment]
@ostephens I hear you, I hear you but that very issue has just been canvassed on the discuss digest. Will try to find the link, but basically it boils down to - you use it, you've learned it, you love it BUT for someone coming to it new (and I have been there with Atom) it is actually quite daunting, mainly because it can Do. So. Much. But for newbies, it is just too much choice. It's like giving someone a toolbox with 50+ things in it when all they wanted was ONE simple screwdriver.

**Comment by weaverbel** [comment]
Ok @ostephens @drjwbaker and others Here is the start of the Atom/nano discussion. http://lists.software-carpentry.org/pipermail/discuss/2017-March/005140.html

**Comment by ostephens** [comment]
Thanks @weaverbel.

I have to admit I don't find the arguments for nano put there overwhelming, but I'm not really bothered enough to argue further. At the end of the dayI want instructors happy with what they are teaching and to use what they find works for them/their students!

The only other point I'd make is that a good text editor is a powerful tool in itself. Maybe what I really would like to see is a lesson which introduces a more powerful text editor such as Atom or Sublime or Notepad++ or ... to librarians

**Comment by drjwbaker** [comment]
I think what I take from this as follows: what is important to LC is that when we use text editors we make the case for text editors. I'm happy for instructors to recommend what they wish in terms of *which* editor.

**Comment by weaverbel** [comment]
Hi @ostephens I agree the case for nano is not great - it was more that the reasons why people objected to Atom because of cognitive load were listed there. I am actually agnostic on this issue - I use gedit and nano and Notepad, and I have taught with people who just LOVE Vim (Gawdelpem) and a colleague here still uses emacs.  I think it might be good to let people start with a basic editor and then tell them - maybe near the end of the session - that there are other, more powerful ones that offer much greater functionality, and then demo one or two features you particularly like about Atom. That way, the keen ones can investigate further post-lesson.. 

What I always do if the editor, e.g. gedit, will launch graphically rather than in the shell window, e.g. nano, is start ANOTHER shell just for the editor - that way, they don't lose their existing work in the shell.  On that, though, I show them how to do `history > commandsused.txt` so they have a copy of all the things they have entered in a session and then can possibly use some of them to build a shell script from later on, or just to refresh their memory. That might be good to include in the shell lesson, as we want to encourage the reuse of code.

**Comment by weaverbel** [comment]
And I agree with @drjwbaker - let's leave the choice open., People teach best what they are comfortable with.

**Comment by drjwbaker** [comment]
Right. Are we agreed(ish) on text editors? Is the template now working? Can we write up a guide on "how to use the template" and close this?

**Comment by weaverbel** [comment]
@drjwbaker I will check with @rgaiacs to see if we are ready to fly. Rainere? Good to go? Thanks for your efforts BTW! There are actually pretty good instructions on the workshop template page James - are you happy that this is enough? https://github.com/swcarpentry/workshop-template Happy to be a sounding board if people get into a tangle with it.

**Comment by jezcope** [comment]
Yup, I'm happy with that as a policy on text editors. I think leaving the decision to the instructor allows the flexibility to choose based on a knowledge of the audience where that's available, and should enable instructors to be more comfortable. It also gives us more scope to try some alternatives and discuss the pros and cons within the community.

For reference, I tend to prefer vi over ssh but generally use emacs with evil-mode, which enables a vi-like style of keyboard interaction. Never let it be said that I'm not perverse about my choice of editor. 😜

**Comment by rgaiacs** [comment]
@weaverbel based on https://github.com/swcarpentry/workshop-template/pull/393, I need (1) changes to syllabus and (2) instructions for the text editor you are going to use. When I have (1) and (2) I can rebase the pull request and merge.

**Comment by weaverbel** [comment]
Hi @rgaiacs 

(1) Is it possible to copy the syllabus info from Tim Dennis's existing customisation? http://www.tim-dennis.com/workshop-template/ That is correct, and you should be able just to copy the markdown from there.  

(2) We are still discussing the text editor issue so I would leave it as the standard SWC one for now and people can customise the instructions (if they want changes) for their own workshops further down the track. I hope that suits @drjwbaker @jezcope  ?

Thanks again Raniere.

**Comment by rgaiacs** [comment]
@weaverbel Thanks. I will work on it before Easter. If you didn't have any news after Easter please get in touch.

**Comment by drjwbaker** [comment]
On (2) using SWC as default for text editors - for now - works for me.

**Comment by rgaiacs** [comment]
@weaverbel I updated the pull request. I emailed you and Kate asking more people to review the changes.

**Comment by weaverbel** [comment]
Thanks @rgaiacs I'll take a look.

**Comment by drjwbaker** [comment]
Where are we with this?

**Comment by jcoliver** [comment]
I'm planning a workshop for early August and it would be great to have this available. I can lend a hand if there's more work needed to get the template off the ground.

**Comment by weaverbel** [comment]
I think the work is done @jcoliver @drjwbaker - when we did workshop template in our last class, LC was available as a flavour of carpentry.

**Comment by drjwbaker** [comment]
It might be, but we don't have documentation on how to use it in place useful for people who want to run an LC workshop (as far as I know). I don't know how to use it..

**Comment by rgaiacs** [comment]
> It might be, but we don't have documentation on how to use it in place useful for people who want to run an LC workshop (as far as I know). I don't know how to use it.

1. Clone https://github.com/swcarpentry/workshop-template
2. At https://github.com/swcarpentry/workshop-template/blob/gh-pages/index.html, change `carpentry: "FIXME"` to `carpentry: "lc"`.
3. Change files at https://github.com/swcarpentry/workshop-template/tree/gh-pages/_includes/lc as you like.

@jcoliver Help with documentation are welcome. :-) Any question, open a issue at https://github.com/swcarpentry/workshop-template/issues.
